plugin/pkg/admission/noderestriction/admission.go

```diff
@@ -655,3 +661,27 @@ func (p *Plugin) admitResourceSlice(nodeName string, a admission.Attributes) err
 
 	return nil
 }
+
+func (p *Plugin) admitCSR(nodeName string, a admission.Attributes) error {
+	// The create request must come from a node with the same name in the CN of the CSR.
+	// Other CSR attributes get checked in CSR validation by the signer.
+	if a.GetOperation() == admission.Create {
+		csr, ok := a.GetObject().(*certapi.CertificateSigningRequest)
+		if !ok {
+			return admission.NewForbidden(a, fmt.Errorf("unexpected type %T", a.GetObject()))
+		}
+		csrBytes := make([]byte, base64.StdEncoding.DecodedLen(len(csr.Spec.Request)))
+		_, err := base64.StdEncoding.Decode(csrBytes, csr.Spec.Request)
+		if err != nil {
+			return admission.NewForbidden(a, fmt.Errorf("unable to base64 decode csr"))
```

Comment:
```suggestion
			return admission.NewForbidden(a, fmt.Errorf("unable to base64 decode csr: %w", err))
```

Reply:
wrapping errors is only useful if you check for that particular error with errors.Is.

Reply:
either way, using `%w` will make the linter stop failing

Reply:
ah didn't realize this was linters doing.